### ISSUE TITLE
hiptensorflow build fails

### ISSUE THREAD

**Issue by Sumenia** [ROCmSoftwarePlatform/hiptensorflow#67]
i followed every step from `tensorflow-build-from-source.md` with a freshly installed Ubuntu 16.04.3 but i keep getting this error when building hiptensorflow 

ERROR: /home/sumenia/hiptensorflow/tensorflow/contrib/framework/BUILD:60:1: undeclared inclusion(s) in rule '//tensorflow/contrib/framework:python/ops/_variable_ops_gpu':
this rule is missing dependency declarations for the following files included by 'tensorflow/contrib/framework/kernels/zero_initializer_op_gpu.cu.cc':
  '/opt/rocm/hcc/lib/clang/7.0.0/include/stdint.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/stddef.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/stdarg.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/__stddef_max_align_t.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/limits.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/x86intrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/ia32intrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/immintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/mmintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/xmmintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/mm_malloc.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/emmintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/f16cintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/pmmintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/tmmintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/smmintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/wmmintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/__wmmintrin_aes.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/__wmmintrin_pclmul.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/clflushoptintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/clwbintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avxintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx2intrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/vpclmulqdqintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/bmiintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/bmi2intrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/lzcntintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/fmaintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512fintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vlintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512bwintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512bitalgintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512cdintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vpopcntdqintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vpopcntdqvlintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vnniintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vlvnniintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512dqintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vlbitalgintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vlbwintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vlcdintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vldqintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512erintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512ifmaintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512ifmavlintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vbmiintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vbmivlintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vbmi2intrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512vlvbmi2intrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/avx512pfintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/pkuintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/vaesintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/gfniintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/rtmintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/xtestintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/shaintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/fxsrintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/xsaveintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/xsaveoptintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/xsavecintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/xsavesintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/cetintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/adxintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/mm3dnow.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/prfchwintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/popcntintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/rdseedintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/ammintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/fma4intrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/xopintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/tbmintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/lwpintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/mwaitxintrin.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/clzerointrin.h'
  '/opt/rocm/hsa/include/hsa/hsa.h'
  '/opt/rocm/hsa/include/hsa/hsa_ext_amd.h'
  '/opt/rocm/hsa/include/hsa/hsa_ext_image.h'
  '/opt/rocm/hcc/lib/clang/7.0.0/include/float.h'.
clang-7.0: warning: /usr/bin/gcc: 'linker' input unused [-Wunused-command-line-argument]
clang-7.0: warning: /usr/bin/gcc: 'linker' input unused [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-isystem external/protobuf/src' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-isystem bazel-out/local_linux-opt/genfiles/external/protobuf/src' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-isystem external/bazel_tools/tools/cpp/gcc3' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-isystem external/eigen_archive' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-isystem bazel-out/local_linux-opt/genfiles/external/eigen_archive' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-isystem external/local_config_cuda/cuda/include' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-isystem bazel-out/local_linux-opt/genfiles/external/local_config_cuda/cuda/include' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote .' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote bazel-out/local_linux-opt/genfiles' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote external/protobuf' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote bazel-out/local_linux-opt/genfiles/external/protobuf' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote external/bazel_tools' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote bazel-out/local_linux-opt/genfiles/external/bazel_tools' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote external/eigen_archive' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote bazel-out/local_linux-opt/genfiles/external/eigen_archive' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote external/local_config_sycl' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote bazel-out/local_linux-opt/genfiles/external/local_config_sycl' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote external/local_config_cuda' [-Wunused-command-line-argument]
clang-7.0: warning: argument unused during compilation: '-iquote bazel-out/local_linux-opt/genfiles/external/local_config_cuda' [-Wunused-command-line-argument]
Target //tensorflow/tools/pip_package:build_pip_package failed to build

**Comment by sunway513** [comment]
@Sumenia , plz add the three lines in the following file:
hiptensorflow/blob/third_party/gpus/crosstool/CROSSTOOL.tpl#L138

```
  cxx_builtin_include_directory: "/opt/rocm/hcc-1.0/compiler/lib/clang/7.0.0/include/"
  cxx_builtin_include_directory: "/opt/rocm/hcc-1.0/lib/clang/7.0.0/include"
  cxx_builtin_include_directory: “/opt/rocm/hsa/include/hsa"

``` 
Then try to configure and rebuild.

**Comment by whchung** [comment]
@Sumenia could you try change the following lines:

https://github.com/ROCmSoftwarePlatform/hiptensorflow/blob/hip/third_party/gpus/crosstool/CROSSTOOL.tpl#L136

and add the following lines:

```
  cxx_builtin_include_directory: "/opt/rocm/hcc/compiler/lib/clang/7.0.0/include/"
  cxx_builtin_include_directory: "/opt/rocm/hcc/lib/clang/7.0.0/include"
```

**Comment by Sumenia** [comment]
thank you both of you for the quick answer !

However, this is still not working 😞 i get new type of errors .... 

I confirm that all hip sample are working perfectly.

this seems the same problem : https://github.com/ROCm-Developer-Tools/HIP/issues/343

do you confirm that current HIP runtime API is incompatible with hiptensorflow ?

tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, Eigen::half, int>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, Eigen::half *>, F = void (*)(hip_impl::Empty_launch_parm, long, Eigen::half *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, Eigen::half *)' for 1st argument
void hipLaunchKernel(
     ^
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, Eigen::half, long long>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, Eigen::half *>, F = void (*)(hip_impl::Empty_launch_parm, long, Eigen::half *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, Eigen::half *)' for 1st argument
void hipLaunchKernel(
     ^
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, float, int>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, float *>, F = void (*)(hip_impl::Empty_launch_parm, long, float *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, float *)' for 1st argument
void hipLaunchKernel(
     ^
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, float, long long>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, float *>, F = void (*)(hip_impl::Empty_launch_parm, long, float *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, float *)' for 1st argument
void hipLaunchKernel(
     ^
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, double, int>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, double *>, F = void (*)(hip_impl::Empty_launch_parm, long, double *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, double *)' for 1st argument
void hipLaunchKernel(
     ^
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, double, long long>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, double *>, F = void (*)(hip_impl::Empty_launch_parm, long, double *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, double *)' for 1st argument
void hipLaunchKernel(
     ^
6 errors generated.
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, Eigen::half, int>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, Eigen::half *>, F = void (*)(hip_impl::Empty_launch_parm, long, Eigen::half *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, Eigen::half *)' for 1st argument
void hipLaunchKernel(
     ^
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, Eigen::half, long long>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, Eigen::half *>, F = void (*)(hip_impl::Empty_launch_parm, long, Eigen::half *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, Eigen::half *)' for 1st argument
void hipLaunchKernel(
     ^
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, float, int>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, float *>, F = void (*)(hip_impl::Empty_launch_parm, long, float *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, float *)' for 1st argument
void hipLaunchKernel(
     ^
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, float, long long>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, float *>, F = void (*)(hip_impl::Empty_launch_parm, long, float *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, float *)' for 1st argument
void hipLaunchKernel(
     ^
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, double, int>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, double *>, F = void (*)(hip_impl::Empty_launch_parm, long, double *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, double *)' for 1st argument
void hipLaunchKernel(
     ^
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:70:5: error: no matching function for call to 'hipLaunchKernel'
    hipLaunchKernel(HIP_KERNEL_NAME(SetZero), dim3(config.block_count), dim3(config.thread_per_block), 0, d.stream(), 
    ^~~~~~~~~~~~~~~
tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc:99:26: note: in instantiation of member function 'tensorflow::functor::UnsortedSegmentSumFunctor<Eigen::GpuDevice, double, long long>::operator()' requested here
TF_CALL_GPU_NUMBER_TYPES(DEFINE_GPU_SPECS);
                         ^
external/local_config_cuda/crosstool/clang/bin/../../../cuda/include/hip/hcc_detail/functional_grid_launch.hpp:142:6: note: candidate function [with Args = <long, double *>, F = void (*)(hip_impl::Empty_launch_parm, long, double *)] not viable: no overload of 'SetZero' matching 'void (*)(hip_impl::Empty_launch_parm, long, double *)' for 1st argument
void hipLaunchKernel(
     ^
6 errors generated.
Died at external/local_config_cuda/crosstool/clang/bin/../../../cuda/bin/hipcc line 501.
ERROR: /home/sumenia/hiptensorflow/tensorflow/core/kernels/BUILD:2131:1: output 'tensorflow/core/kernels/_objs/segment_reduction_ops_gpu/tensorflow/core/kernels/segment_reduction_ops_gpu.cu.pic.o' was not created.
ERROR: /home/sumenia/hiptensorflow/tensorflow/core/kernels/BUILD:2131:1: not all outputs were created or valid.
Target //tensorflow/tools/pip_package:build_pip_package failed to build

**Comment by sunway513** [comment]
Could you try to cherry-pick the following commit?
`git cherry-pick 417dfce2180a49e1bbe38db10025b5da4101c153`

**Comment by Sumenia** [comment]
awesome, this is working ! thank you for your help 
should i commit and PR those modification ?

**Comment by sunway513** [comment]
Great! Please submit a PR to the `hip-develop` branch, appreciate your contributions!

**Comment by Sumenia** [comment]
i don`t know if its related to the modification i had to do i order to compile but none convnet-benchmarks works. 

benchmark_alexnet.py / benchmark_googlenet.py / benchmark_overfeat.py i get :
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  No device code available for function: _ZN5Eigen8internal15EigenMetaKernelINS_15TensorEvaluatorIKNS_14TensorAssignOpINS_9TensorMapINS_6TensorIfLi1ELi1EiEELi16ENS_11MakePointerEEEKNS_18TensorCwiseUnaryOpINS0_12scalar_rightIffNS0_18scalar_quotient_opIffEEEEKNS4_INS5_IKfLi1ELi1EiEELi16ES7_EEEEEENS_9GpuDeviceEEEiEEvT_T0_
Aborted (core dumped)
```
benchmark_vgg.py : 
```
I tensorflow/core/common_runtime/gpu/gpu_device.cc:892] Found device 0 with properties: 
name: Device 687f
major: 3 minor: 0 memoryClockRate (GHz) 1.59
pciBusID 0000:0c:00.0
Total memory: 7.98GiB
Free memory: 7.73GiB
I tensorflow/core/common_runtime/gpu/gpu_device.cc:913] DMA: 0 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] 0:   Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:972] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Device 687f, pci bus id: 0000:0c:00.0)
I tensorflow/core/kernels/conv_ops.cc:604] running auto-tune for Convolve
MIOpen Error: /home/sumenia/MIOpen/src/hip/handlehip.cpp:68: Memory not available to allocate buffer: 822083584
MIOpen Error: /home/sumenia/MIOpen/src/include/miopen/errors.hpp:97: Dereferencing nullptr
```

i confirm i did `export HIP_VISIBLE_DEVICES=0`

do you think its related or should i open another issue for this ?
